### ISSUE TITLE
Invalid `\n` escaping by the driver

### ISSUE THREAD

**Issue by vitaly-t** [vitaly-t/pg-promise#253]
The underlying driver [node-postgres] does invalid escaping for `\n` for type `text[]`.

This issue comes up only when you use initialization option `pgFormatting`, or when such a type is being passed in via a [PreparedStatement] or [ParameterizedQuery].

This issues was logged against the driver: https://github.com/brianc/node-postgres/issues/1147

**Important:** It doesn't occur when using the default/string query formatting provided by `pg-promise`.

[node-postgres]:https://github.com/brianc/node-postgres
[PreparedStatement]:http://vitaly-t.github.io/pg-promise/PreparedStatement.html
[ParameterizedQuery]:http://vitaly-t.github.io/pg-promise/ParameterizedQuery.html

**Comment by abenhamdine** [comment]
This has been fixed in pg@6.1.1

**Comment by vitaly-t** [comment]
@abenhamdine yeah, I noticed that :wink:

It is one of the commits I'm planning for `pg-core` when I get around to finishing it.

**Comment by AlJohri** [comment]
@vitaly-t what is `pg-core`?

**Comment by vitaly-t** [comment]
@AlJohri an unfinished attempt at branching out an independent version of the `node-postgres` driver.

**Comment by vitaly-t** [comment]
@AlJohri The things is, some time ago I did that, created an independent module form `node-postgres`, and it all worked fine.

The reason I eventually axed it was that tests in the existing `node-portgres` are horrific.... they are very much custom-written, and they do not work on Windows at all.

I had to rewrite a substential part of tests and dump many of them just to make the test work on on all platforms. I wasn't impressed by the reasult, and getting it to 100% didn't seem realistic at the time.

I still have it backed up in a private branch, so may revive it later on.

The bottom line is, I found everything that occurred with `node-postgres` v5.1 a bit of a disappointment, and mostly compromizing on its quality rather than improvement. But I'm back to ranting about it :)

**Comment by abenhamdine** [comment]
> So, pg-core is still in the plans.
Glad to read this 😃 
Because no activity in https://github.com/brianc/node-postgres since Dec 16th... 😞

**Comment by vitaly-t** [comment]
Yes, because it is not desperate at present, the `node-postgrs` driver didn't get any critical updates since v5.1, which remains to be its best version. I think we are good for some time forward :wink:

**Comment by abenhamdine** [comment]
yes, 42 commits since 5.1 : https://github.com/brianc/node-postgres/compare/v5.1.0...master
including ~20 readme or changelog changes, and several commits related to the new pool.
So probably less than 10 commits to backport if you are willing to..

**Comment by vitaly-t** [comment]
Much less, by my account, about 3 that are minor improvements, the rest is unimportant at all.

I'm planning to have this `pg-core` thing working for version 6.0.0 of the library.

**Comment by abenhamdine** [comment]
I would be glad to help on this, unfortunately I have no spare time at the moment...
Perhaps in a few weeks, when I will have a new developper in the team.

**Comment by vitaly-t** [comment]
Time is everything. I will post an update here when I get around to `pg-core` again. Like i said, its biggest issue will be rewriting most of the tests.

**Comment by vitaly-t** [comment]
I'm closing it now, because there is no point keeping just one minor issue here. There are several minor issues that have been fixed in the driver since version 5.1, and those all will need to be addressed, somehow.

I am particularly concerned about this fresh one: https://github.com/brianc/node-postgres/issues/1286

Some sort of a merge / branching is unavoidable later on, that's for sure.

**Comment by vitaly-t** [comment]
There is [branch 6.x](https://github.com/vitaly-t/pg-promise/tree/6.x) available now, for anyone who is interested :wink:
